File: redome-front/src/app/shared/dto/doador.cordao.internacional.dto.ts
import {Registro} from "../dominio/registro";
import {StatusDoadorDTO} from "./status.doador.internacional.dto";
import {LocusExame} from "../../paciente/cadastro/exame/locusexame";
import {ValorGenotipoDTO} from "./valor.genotipo.dto";
import {BaseEntidade} from "../base.entidade";

export class DoadorCordaoInternacionalDto extends BaseEntidade {

   private _id:number;

	private _grid:string;

	private _idade:number;

	private _peso:number;

	private _sexo:string;

	private _abo:string;

	private _dataNascimento:Date;

	private _statusDoador:StatusDoadorDTO;

	private _dataRetornoInatividade:Date;

	private _idRegistro:String;

	private _registroOrigem:Registro;

	private _registroPagamento:Registro;

	private _cadastradoEmdis:Boolean;

	private _quantidadeTotalTCN:number;

	private _quantidadeTotalCD34:number;

	private _volume:number;

	private _rmrAssociado:number;

	private _tipoDoador: number;

	private _locusExames: LocusExame[] = [];

	private _valoresGenotipo: ValorGenotipoDTO[];

	private _ressalvas: string[];

	constructor(tipoDoador: number) {
		super();
		this._tipoDoador = tipoDoador;
	}

    /**
     * Getter id
     * @return {number}
     */
	public get id(): number {
		return this._id;
	}

    /**
     * Getter grid
     * @return {string}
     */
	public get grid(): string {
		return this._grid;
	}

    /**
     * Getter idade
     * @return {number}
     */
	public get idade(): number {
		return this._idade;
	}

    /**
     * Getter peso
     * @return {number}
     */
	public get peso(): number {
		return this._peso;
	}

    /**
     * Getter sexo
     * @return {string}
     */
	public get sexo(): string {
		return this._sexo;
	}

    /**
     * Getter abo
     * @return {string}
     */
	public get abo(): string {
		return this._abo;
	}

    /**
     * Getter dataNascimento
     * @return {Date}
     */
	public get dataNascimento(): Date {
		return this._dataNascimento;
	}

    /**
     * Getter statusDoador
     * @return {StatusDoadorDTO}
     */
	public get statusDoador(): StatusDoadorDTO {
		return this._statusDoador;
	}

    /**
     * Getter dataRetornoInatividade
     * @return {Date}
     */
	public get dataRetornoInatividade(): Date {
		return this._dataRetornoInatividade;
	}

    /**
     * Getter idRegistro
     * @return {String}
     */
	public get idRegistro(): String {
		return this._idRegistro;
	}

    /**
     * Getter registroOrigem
     * @return {Registro}
     */
	public get registroOrigem(): Registro {
		return this._registroOrigem;
	}

    /**
     * Getter registroPagamento
     * @return {Registro}
     */
	public get registroPagamento(): Registro {
		return this._registroPagamento;
	}

    /**
     * Getter cadastradoEmdis
     * @return {Boolean}
     */
	public get cadastradoEmdis(): Boolean {
		return this._cadastradoEmdis;
	}

    /**
     * Getter quantidadeTotalTCN
     * @return {number}
     */
	public get quantidadeTotalTCN(): number {
		return this._quantidadeTotalTCN;
	}

    /**
     * Getter quantidadeTotalCD34
     * @return {number}
     */
	public get quantidadeTotalCD34(): number {
		return this._quantidadeTotalCD34;
	}

    /**
     * Getter volume
     * @return {number}
     */
	public get volume(): number {
		return this._volume;
	}

    /**
     * Getter rmrAssociado
     * @return {number}
     */
	public get rmrAssociado(): number {
		return this._rmrAssociado;
	}

    /**
     * Getter tipoDoador
     * @return {String}
     */
	public get tipoDoador(): number {
		return this._tipoDoador;
	}

    /**
     * Setter id
     * @param {number} value
     */
	public set id(value: number) {
		this._id = value;
	}

    /**
     * Setter grid
     * @param {string} value
     */
	public set grid(value: string) {
		this._grid = value;
	}

    /**
     * Setter idade
     * @param {number} value
     */
	public set idade(value: number) {
		this._idade = value;
	}

    /**
     * Setter peso
     * @param {number} value
     */
	public set peso(value: number) {
		this._peso = value;
	}

    /**
     * Setter sexo
     * @param {string} value
     */
	public set sexo(value: string) {
		this._sexo = value;
	}

    /**
     * Setter abo
     * @param {string} value
     */
	public set abo(value: string) {
		this._abo = value;
	}

    /**
     * Setter dataNascimento
     * @param {Date} value
     */
	public set dataNascimento(value: Date) {
		this._dataNascimento = value;
	}

    /**
     * Setter statusDoador
     * @param {StatusDoadorDTO} value
     */
	public set statusDoador(value: StatusDoadorDTO) {
		this._statusDoador = value;
	}

    /**
     * Setter dataRetornoInatividade
     * @param {Date} value
     */
	public set dataRetornoInatividade(value: Date) {
		this._dataRetornoInatividade = value;
	}

    /**
     * Setter idRegistro
     * @param {String} value
     */
	public set idRegistro(value: String) {
		this._idRegistro = value;
	}

    /**
     * Setter registroOrigem
     * @param {Registro} value
     */
	public set registroOrigem(value: Registro) {
		this._registroOrigem = value;
	}

    /**
     * Setter registroPagamento
     * @param {Registro} value
     */
	public set registroPagamento(value: Registro) {
		this._registroPagamento = value;
	}

    /**
     * Setter cadastradoEmdis
     * @param {Boolean} value
     */
	public set cadastradoEmdis(value: Boolean) {
		this._cadastradoEmdis = value;
	}

    /**
     * Setter quantidadeTotalTCN
     * @param {number} value
     */
	public set quantidadeTotalTCN(value: number) {
		this._quantidadeTotalTCN = value;
	}

    /**
     * Setter quantidadeTotalCD34
     * @param {number} value
     */
	public set quantidadeTotalCD34(value: number) {
		this._quantidadeTotalCD34 = value;
	}

    /**
     * Setter volume
     * @param {number} value
     */
	public set volume(value: number) {
		this._volume = value;
	}

    /**
     * Setter rmrAssociado
     * @param {number} value
     */
	public set rmrAssociado(value: number) {
		this._rmrAssociado = value;
	}

	get locusExames(): LocusExame[] {
		return this._locusExames;
	}

	set locusExames(value: LocusExame[]) {
		this._locusExames = value;
	}

	get valoresGenotipo(): ValorGenotipoDTO[] {
		return this._valoresGenotipo;
	}

	set valoresGenotipo(value: ValorGenotipoDTO[]) {
		this._valoresGenotipo = value;
	}

	get ressalvas(): string[] {
		return this._ressalvas;
	}

	set ressalvas(value: string[]) {
		this._ressalvas = value;
	}

	jsonToEntity(res: any): DoadorCordaoInternacionalDto {
		return null;
	}

}

File: redome/src/main/java/br/org/cancer/modred/controller/dto/doador/DoadorCordaoInternacionalDTO.java
package br.org.cancer.modred.controller.dto.doador;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

/**
 * Dto para DoadorInternacional.
 * 
 * @author Filipe Paes
 *
 */

public class DoadorCordaoInternacionalDTO  implements Serializable {

	private static final long serialVersionUID = -3851323934063893734L;
	
	private Long tipoDoador;

	private Long rmrAssociado;
	
	private Long id;
	
	private LocalDate dataCadastro;
	
	private RegistroDTO registroOrigem;
	
	private String sexo;
	
	private String abo;
	
	private LocalDate dataNascimento;
	
	private LocalDateTime dataAtualizacao;
	
	private StatusDoadorDTO statusDoador;
	
	private LocalDate dataRetornoInatividade;
	
	private BigDecimal peso;
	
	private Integer idade;
	
	private String grid;

	private String idRegistro;
	
	private RegistroDTO registroPagamento;

	private Boolean cadastradoEmdis;
	
	private List<String> ressalvas;
	
	private List<LocusExameDTO> locusExames;
	
	private List<ValorGenotipoDTO> valoresGenotipo;
	
	private BigDecimal quantidadeTotalTCN;

	private BigDecimal quantidadeTotalCD34;

	private BigDecimal volume;

	/**
	 * @return the tipoDoador
	 */
	public Long getTipoDoador() {
		return tipoDoador;
	}

	/**
	 * @param tipoDoador the tipoDoador to set
	 */
	public void setTipoDoador(Long tipoDoador) {
		this.tipoDoador = tipoDoador;
	}

	/**
	 * @return the rmrAssociado
	 */
	public Long getRmrAssociado() {
		return rmrAssociado;
	}

	/**
	 * @param rmrAssociado the rmrAssociado to set
	 */
	public void setRmrAssociado(Long rmrAssociado) {
		this.rmrAssociado = rmrAssociado;
	}

	/**
	 * @return the id
	 */
	public Long getId() {
		return id;
	}

	/**
	 * @param id the id to set
	 */
	public void setId(Long id) {
		this.id = id;
	}

	/**
	 * @return the dataCadastro
	 */
	public LocalDate getDataCadastro() {
		return dataCadastro;
	}

	/**
	 * @param dataCadastro the dataCadastro to set
	 */
	public void setDataCadastro(LocalDate dataCadastro) {
		this.dataCadastro = dataCadastro;
	}

	/**
	 * @return the registroOrigem
	 */
	public RegistroDTO getRegistroOrigem() {
		return registroOrigem;
	}

	/**
	 * @param registroOrigem the registroOrigem to set
	 */
	public void setRegistroOrigem(RegistroDTO registroOrigem) {
		this.registroOrigem = registroOrigem;
	}

	/**
	 * @return the sexo
	 */
	public String getSexo() {
		return sexo;
	}

	/**
	 * @param sexo the sexo to set
	 */
	public void setSexo(String sexo) {
		this.sexo = sexo;
	}

	/**
	 * @return the abo
	 */
	public String getAbo() {
		return abo;
	}

	/**
	 * @param abo the abo to set
	 */
	public void setAbo(String abo) {
		this.abo = abo;
	}

	/**
	 * @return the dataNascimento
	 */
	public LocalDate getDataNascimento() {
		return dataNascimento;
	}

	/**
	 * @param dataNascimento the dataNascimento to set
	 */
	public void setDataNascimento(LocalDate dataNascimento) {
		this.dataNascimento = dataNascimento;
	}

	/**
	 * @return the dataAtualizacao
	 */
	public LocalDateTime getDataAtualizacao() {
		return dataAtualizacao;
	}

	/**
	 * @param dataAtualizacao the dataAtualizacao to set
	 */
	public void setDataAtualizacao(LocalDateTime dataAtualizacao) {
		this.dataAtualizacao = dataAtualizacao;
	}

	/**
	 * @return the statusDoador
	 */
	public StatusDoadorDTO getStatusDoador() {
		return statusDoador;
	}

	/**
	 * @param statusDoador the statusDoador to set
	 */
	public void setStatusDoador(StatusDoadorDTO statusDoador) {
		this.statusDoador = statusDoador;
	}

	/**
	 * @return the dataRetornoInatividade
	 */
	public LocalDate getDataRetornoInatividade() {
		return dataRetornoInatividade;
	}

	/**
	 * @param dataRetornoInatividade the dataRetornoInatividade to set
	 */
	public void setDataRetornoInatividade(LocalDate dataRetornoInatividade) {
		this.dataRetornoInatividade = dataRetornoInatividade;
	}

	/**
	 * @return the peso
	 */
	public BigDecimal getPeso() {
		return peso;
	}

	/**
	 * @param peso the peso to set
	 */
	public void setPeso(BigDecimal peso) {
		this.peso = peso;
	}

	/**
	 * @return the idade
	 */
	public Integer getIdade() {
		return idade;
	}

	/**
	 * @param idade the idade to set
	 */
	public void setIdade(Integer idade) {
		this.idade = idade;
	}

	/**
	 * @return the grid
	 */
	public String getGrid() {
		return grid;
	}

	/**
	 * @param grid the grid to set
	 */
	public void setGrid(String grid) {
		this.grid = grid;
	}

	/**
	 * @return the idRegistro
	 */
	public String getIdRegistro() {
		return idRegistro;
	}

	/**
	 * @param idRegistro the idRegistro to set
	 */
	public void setIdRegistro(String idRegistro) {
		this.idRegistro = idRegistro;
	}

	/**
	 * @return the registroPagamento
	 */
	public RegistroDTO getRegistroPagamento() {
		return registroPagamento;
	}

	/**
	 * @param registroPagamento the registroPagamento to set
	 */
	public void setRegistroPagamento(RegistroDTO registroPagamento) {
		this.registroPagamento = registroPagamento;
	}

	/**
	 * @return the cadastradoEmdis
	 */
	public Boolean getCadastradoEmdis() {
		return cadastradoEmdis;
	}

	/**
	 * @param cadastradoEmdis the cadastradoEmdis to set
	 */
	public void setCadastradoEmdis(Boolean cadastradoEmdis) {
		this.cadastradoEmdis = cadastradoEmdis;
	}


	/**
	 * @return the quantidadeTotalTCN
	 */
	public BigDecimal getQuantidadeTotalTCN() {
		return quantidadeTotalTCN;
	}

	/**
	 * @param quantidadeTotalTCN the quantidadeTotalTCN to set
	 */
	public void setQuantidadeTotalTCN(BigDecimal quantidadeTotalTCN) {
		this.quantidadeTotalTCN = quantidadeTotalTCN;
	}

	/**
	 * @return the quantidadeTotalCD34
	 */
	public BigDecimal getQuantidadeTotalCD34() {
		return quantidadeTotalCD34;
	}

	/**
	 * @param quantidadeTotalCD34 the quantidadeTotalCD34 to set
	 */
	public void setQuantidadeTotalCD34(BigDecimal quantidadeTotalCD34) {
		this.quantidadeTotalCD34 = quantidadeTotalCD34;
	}

	/**
	 * @return the volume
	 */
	public BigDecimal getVolume() {
		return volume;
	}

	/**
	 * @param volume the volume to set
	 */
	public void setVolume(BigDecimal volume) {
		this.volume = volume;
	}

	/**
	 * @return the ressalvas
	 */
	public List<String> getRessalvas() {
		return ressalvas;
	}

	/**
	 * @param ressalvas the ressalvas to set
	 */
	public void setRessalvas(List<String> ressalvas) {
		this.ressalvas = ressalvas;
	}
	
	/**
	 * @return the locusExames
	 */
	public List<LocusExameDTO> getLocusExames() {
		return locusExames;
	}

	/**
	 * @param locusExames the locusExames to set
	 */
	public void setLocusExames(List<LocusExameDTO> locusExames) {
		this.locusExames = locusExames;
	}

	/**
	 * @return the valoresGenotipo
	 */
	public List<ValorGenotipoDTO> getValoresGenotipo() {
		return valoresGenotipo;
	}

	/**
	 * @param valoresGenotipo the valoresGenotipo to set
	 */
	public void setValoresGenotipo(List<ValorGenotipoDTO> valoresGenotipo) {
		this.valoresGenotipo = valoresGenotipo;
	}
		
	
}
